packages/react-reconciler/src/ReactFiberHooks.new.js

```diff
@@ -2471,7 +2471,7 @@ const HooksDispatcherOnRerender: Dispatcher = {
   useDeferredValue: rerenderDeferredValue,
   useTransition: rerenderTransition,
   useMutableSource: updateMutableSource,
-  useSyncExternalStore: mountSyncExternalStore,
+  useSyncExternalStore: updateSyncExternalStore,
```

Comment:
Nice catch.

This change needs to be made to `packages/react-reconciler/src/ReactFiberHooks.old.js` also. (You can run `yarn replace-fork` to do this automatically.)